File: solutions/java/src/trafficsignalsystem/GreenState.java
package trafficsignalsystem;

public class GreenState implements SignalState {
    @Override
    public void handle(TrafficLight light, TrafficSignalController controller, Direction direction) {
        System.out.println("Direction: " + direction + " | State: GREEN");
        int duration = controller.getSignalDuration(direction, this);
        controller.scheduleStateChange(light, direction, new YellowState(), duration);
    }
    @Override
    public String getName() { return "GREEN"; }
} 

File: solutions/typescript/src/TrafficSignalSystem/SignalState/GreenState.ts
import { DirectionEnum } from "../Direction";
import TrafficLight from "../TrafficLight";
import TrafficSignalController from "../TrafficSignalController";
import SignalState from "./SignalState";
import YellowState from "./YellowState";

export default class GreenState implements SignalState {
    handle(
        light: TrafficLight,
        controller: TrafficSignalController,
        direction: DirectionEnum,
    ) {
        console.log("Direction: " + direction + " | State: GREEN");
        const duration = controller.getSignalDuration(direction, this);
        controller.scheduleStateChange(
            light,
            direction,
            new YellowState(),
            duration,
        );
    }

    getName() {
        return "GREEN";
    }
}
